src/models/image-map.ts

```diff
@@ -417,16 +446,69 @@ export const localAssetsImagesHandler: IImageHandler = {
     return url ? url.startsWith("assets/") || url.startsWith("curriculum/") : false;
   },
 
-  async store(url: string) {
+  async store(url: string, curriculumBaseUrl?: string, unitCodeMap?: any, options?: IImageHandlerStoreOptions) {
+    const unitCode = urlParams.unit && isValidHttpUrl(urlParams.unit)
+                     ? getUnitCodeFromUrl(urlParams.unit)
+                     : urlParams.unit
+                       ? urlParams.unit
+                       : "sas";
+    const curriculumBranch = urlParams.unit && isValidHttpUrl(urlParams.unit)
+                               ? getCurriculumBranchFromUrl(urlParams.unit)
+                               : undefined;
+    // Legacy curriculum units lived in the defunct "curriculum" directory in subfolders
+    // named after the unit title, e.g. "curriculum/stretching-and-shrinking". References
+    // to files in those directories will need to be converted to use the new file structure
+    // which uses the unit code as the directory name, e.g. "sas".
+    const urlPieces = url.split("curriculum/");
+    const urlUnitDir = urlPieces[1] ? urlPieces[1].replace(/\/.*/, "") : "";
+    const newUnitDir = urlUnitDir !== "" ? unitCodeMap[urlUnitDir] : "";
                     // convert original curriculum image paths
-    const _url = url.replace("assets/curriculum", "curriculum")
+    const _url = url.replace("assets/curriculum/", "")
                     // convert original drawing tool stamp paths
-                    .replace("assets/tools/drawing-tool/stamps",
-                             "curriculum/moving-straight-ahead/stamps");
-    return { contentUrl: _url, displayUrl: getAssetUrl(_url), success: true  };
+                    .replace("assets/tools/drawing-tool/stamps", "msa/stamps")
+                    // remove curriculum/ from path since that directory no longer exists
+                    .replace("curriculum/", "")
+                    // convert legacy directory names to new ones
+                    .replace(urlUnitDir, newUnitDir);
```

Comment:
This approach might cause some problems. I figure some of the unit directories are common words. So they might get matched in weird ways. Using the pattern approach above I think would eliminate this issue.